app/src/main/java/kr/ac/konkuk/gdsc/plantory/presentation/plant/AddPlantViewModel.kt

```diff
@@ -19,29 +39,104 @@ class AddPlantViewModel : ViewModel() {
+    val plantRegisterItem: StateFlow<PlantRegisterItem> get() = _plantRegisterItem
+
+    private val _postRegisterPlantState = MutableStateFlow<UiState<Unit>>(UiState.Loading)
+    val postRegisterPlantState: StateFlow<UiState<Unit>> = _postRegisterPlantState.asStateFlow()
+
+    private val _plantImage = MutableStateFlow<Uri?>(null)
+    val plantImage: StateFlow<Uri?> = _plantImage.asStateFlow()
+
+    private var imageRequestBody: ContentUriRequestBody? = null
 
     init {
         _currentYear.value = calendar.get(Calendar.YEAR)
         _currentMonth.value = calendar.get(Calendar.MONTH)
         _currentDay.value = calendar.get(Calendar.DAY_OF_MONTH)
+        _plantRegisterItem.value = PlantRegisterItem(
+            "",
+            "",
+            "",
+            "${currentYear.value}-${formatDateToAddZero(currentMonth.value+1)}-${formatDateToAddZero(currentDay.value)}",
+            "${currentYear.value}-${formatDateToAddZero(currentMonth.value+1)}-${formatDateToAddZero(currentDay.value)}",
+            )
+    }
+
+    fun postRegisterPlant() {
+        viewModelScope.launch {
+            _postRegisterPlantState.value = UiState.Loading
+            val (request, image) = createRequestBody(plantRegisterItem.value)
+
+            plantRepository.postRegisterPlant(request, image).onSuccess { response ->
+                _postRegisterPlantState.value = UiState.Success(response)
+                Timber.e("성공 $response")
+            }.onFailure { t ->
+                if (t is HttpException) {
+                    val errorResponse = t.response()?.errorBody()?.string()
+                    Timber.e("HTTP 실패: $errorResponse")
+                }
+                Timber.e("${t.message}")
+                _postRegisterPlantState.value = UiState.Failure("${t.message}")
+            }
+        }
     }
 
-    fun updatePlantInfo(newPlantInfo: PlantInfo) {
-        _plantInfo.value = newPlantInfo
+    fun updateRequestBody(requestBody: ContentUriRequestBody) {
+        this.imageRequestBody = requestBody
+    }
+
+    private fun createRequestBody(
+        plant: PlantRegisterItem
+    ): Pair<HashMap<String, RequestBody>, MultipartBody.Part?> {
+        val imageFormData = imageRequestBody?.toFormData()
+        val plantDto = RequestPostRegisterPlantDto(
+            plantInformationId = findPlantInformationId(plant.species),
+            nickname = plant.nickname,
+            shortDescription = plant.shortDescription,
+            birthDate = plant.birthDate,
+            lastWaterDate = plant.lastWaterDate
+        )
+
+
+        val plantInfoJsonString =
+            Json.encodeToString(RequestPostRegisterPlantDto.serializer(), plantDto)
+
+        val plainTextRequestBody = hashMapOf(
+            "request" to plantInfoJsonString.toRequestBody("application/json".toMediaTypeOrNull())
+        )
+        return Pair(plainTextRequestBody, imageFormData)
```

Comment:
```suggestion
        return plainTextRequestBody to imageFormData
```
이런식으로도 적을 수 있습니다 (권장은 아님 그냥 이런것도 있다구요)